src/output-json-alert.c

```diff
@@ -169,6 +171,32 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
     jb_close(js);
 }
 
+static void AlertJsonReference(
+        AlertJsonOutputCtx *json_output_ctx, const PacketAlert *pa, JsonBuilder *jb)
+{
+    if (!pa->s->references) {
+        return;
+    }
+
+    const DetectReference *kv = pa->s->references;
+    jb_open_array(jb, "references");
+    while (kv) {
+        BUG_ON(kv->key_len > REFERENCE_SYSTEM_NAME_MAX);
+        BUG_ON(kv->reference_len > REFERENCE_CONTENT_NAME_MAX);
```

Comment:
Q1: Is this possible?
Q2: Is this as big an issue that should lead Suricata to exit?

Reply:
Good catch -- they're no longer needed as the bounds checking occurs when parsing the reference config values.